infra/charts/feast/charts/feast-jobservice/templates/deployment.yaml

```diff
@@ -0,0 +1,117 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ template "feast-jobservice.fullname" . }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app: {{ template "feast-jobservice.name" . }}
+    component: jobservice
+    chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+spec:
+  replicas: {{ .Values.replicaCount }}
+  selector:
+    matchLabels:
+      app: {{ template "feast-jobservice.name" . }}
+      component: jobservice
+      release: {{ .Release.Name }}
+  template:
+    metadata:
+      {{- if .Values.prometheus.enabled }}
+      annotations:
+        prometheus.io/path: /metrics
+        prometheus.io/port: "{{ .Values.service.http.targetPort }}"
+        prometheus.io/scrape: "true"
+      {{- end }}
+      labels:
+        app: {{ template "feast-jobservice.name" . }}
+        component: jobservice
+        release: {{ .Release.Name }}
+      {{- if .Values.podLabels }}
+        {{ toYaml .Values.podLabels | nindent 8 }}
+      {{- end }}
+    spec:
+      {{- with .Values.nodeSelector }}
+      nodeSelector:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+
+      {{- if .Values.gcpServiceAccount.enabled }}
```

Comment:
Do we need to hardcode the GCP specific fields here still? I think we want to move away from this (if we support AWS)

Reply:
@woop we should probably handle that in another PR to get this one unblocked.
- All services (core, serving, jupyter) have the gcp-specific fields, so we need to figure out the overall solution here, not just for job service
- Looks like for AWS @oavdeev wrote the terraform script that uses envOverrides to override some environment variables, and he won't need to hardcode any AWS-specific stuff. However, this approach can't be used readily, since we don't have terraform for GCP deployment & some of these gcp fields are used outside env variables right now. Therefore I don't have a clear alternative to move these variables to.
- It won't affect AWS deployment in the short term, so we can temporarily leave it here

Let me know what you think!

Reply:
Sure, lets go ahead with cleaning this up in another PR.